go/kbfs/libkbfs/favorites.go

```diff
@@ -630,10 +672,22 @@ func (f *Favorites) RefreshCache(ctx context.Context) {
 		ctx:     context.Background(),
 	}
 	f.wg.Add(1)
+
+	if async {
+		favResult, err := f.config.KBPKI().FavoriteList(ctx)
+		if err != nil {
+			f.log.CDebugf(ctx, "Failed to refresh cached Favorites: %+v", err)
+			f.wg.Done()
+			<-f.refreshWaiting
+			return
+		}
+		req.favResult = &favResult
```

Comment:
Shouldn't we `f.wg.Done()` when we're successful?

Reply:
No, because we've sent the request into the queue when we're successful, and the main thread will process the request and then call that. See lines 503-508:

```func (f *Favorites) loop() {
	for req := range f.reqChan {
		f.handleReq(req)
		f.wg.Done()
	}
}```

Should I add a comment to this effect?

Reply:
Yes please.

Reply:
done.